### PR TITLE
RuntimeType serialization

### PR DESCRIPTION
**Pull request by ReubenBond**
The first commit ensures that the fallback serializer is not available for most of the tests in BuiltInSerializerTests - most tests should pass without the fallback.

The second commit registers a serializer for `RuntimeType` (the runtime type of `Type` 😉), aligns the serializer with the deserializer, and adds a test.

### CODE REVIEW

File: src/Orleans/Serialization/BuiltInTypes.cs [hunk 1]
```diff
@@ -1637,7 +1637,7 @@ internal static object CopyDateTimeOffset(object obj)
 
         internal static void SerializeType(object obj, BinaryTokenStreamWriter stream, Type expected)
         {
-            stream.Write(((Type)obj).OrleansTypeName());
+            stream.Write(((Type)obj).OrleansTypeKeyString());
```
Comment (sergeybykov) [inline src/Orleans/Serialization/BuiltInTypes.cs]: This won't break backward compatibility with data serialized and persisted previously, right?

Comment (ReubenBond) [inline src/Orleans/Serialization/BuiltInTypes.cs]: It won't break backward compatibility in any case which works today (where `OrleansTypeKeyString()` & `OrleansTypeName()` just happen to produce the same result, eg:
```C#
> typeof(Grain).OrleansTypeName()
"Orleans.Grain"
> typeof(Grain).OrleansTypeKeyString()
"Orleans.Grain" // Simple (non-generic, non-array, etc) types not in "System" will be equal.
> typeof(int).OrleansTypeName()
"System.Int32"
> typeof(int).OrleansTypeKeyString()
"Int32" // Types in System wouldn't work.
```

If you look at `DeserializeType` you'll see it calls `ResolveTypeName` which looks for the type in `types`. If you look for where `types` gets modified, you'll see that the keys are generated using `OrleansTypeKeyString()` and not `OrleansTypeName()` - so the current serialize method is wrong to use `OrleansTypeName()`.

Comment (ReubenBond) [inline src/Orleans/Serialization/BuiltInTypes.cs]: This PR just makes these two consistent. The types still need to be 'registered' in order for serialization to succeed. Ideally, we would remove that requirement.

**Comment by jdom** [comment]
LGTM. Unless @sergeybykov has a concern, I'll merge it

**Pull request merged**